lib/code_corps/onboarding/user_transition.ex

```diff
@@ -0,0 +1,10 @@
+defmodule CodeCorps.Onboarding.UserTransition do
+
+  def next(current, nil), do: nil
+  def next("signed_up", "edit_profile"), do: {:ok, "edited_profile"}
+  def next("edited_profile", "select_categories"), do: {:ok, "selected_categories"}
+  def next("selected_categories", "select_roles"), do: {:ok, "selected_roles"}
+  def next("selected_roles", "select_skills"), do: {:ok, "selected_skills"}
+  def next(current, invalid_transition), do: {:error, "invalid transition #{invalid_transition} from #{current}"}
```

Comment:
It should be as simple as testing the return value of `next` a bunch of times. I personally don't think there's a lot of value in these sorts of tests (code is very simple and you have to maintain an additional test file). If someone else wants to take them on at a later time I can create an issue for it.